Draft ExtensionArray post

TomAugspurger opened the pull request.

TomAugspurger committed:
Draft ExtensionArray post

mrocklin commented on _posts/2019-01-16-dask-extension-arrays.md:
```diff
@@ -0,0 +1,210 @@
+
+Newer versions of pandas allow third-party libraries to write custom extension
+arrays. These arrays can be placed inside a DataFrame or Series, and work
+just as well as any extension arary defined within pandas itself. However,
```
```suggestion
just as well as any extension array defined within pandas itself. However,
```

mrocklin commented on _posts/2019-01-16-dask-extension-arrays.md:
```diff
@@ -0,0 +1,210 @@
+
+Pandas 0.23 introduced the [`ExtensionArray`][EA], a way to store things other
+than a simple NumPy array in a DataFrame or Series. Internally pandas uses this
+for data types that aren't handled natively by NumPy (datetimes with timezones,
```
```suggestion
for data types that aren't handled natively by NumPy like datetimes with timezones,
```

mrocklin commented on _posts/2019-01-16-dask-extension-arrays.md:
```diff
@@ -0,0 +1,210 @@
+Pandas 0.23 introduced the [`ExtensionArray`][EA], a way to store things other
+than a simple NumPy array in a DataFrame or Series. Internally pandas uses this
+for data types that aren't handled natively by NumPy (datetimes with timezones,
+Categorical, and nullable integer arrays, to name a few).
```
```suggestion
Categorical, and nullable integer arrays (new!).
```

mrocklin commented on _posts/2019-01-16-dask-extension-arrays.md:
```diff
@@ -0,0 +1,210 @@
+
+> Built with the broader community
+
+At the start of Dask, the developers *could* have gone off an re-written pandas
```
```suggestion
At the start of Dask, the developers *could* have gone off and re-written pandas
```

mrocklin commented on _posts/2019-01-16-dask-extension-arrays.md:
```diff
@@ -0,0 +1,210 @@
+If you're writing an ExtensionArray, make sure to add it to the [pandas
+ecosystem](http://pandas.pydata.org/pandas-docs/version/0.24/ecosystem.html#extension-data-types)
+page, and register it with Dask.
+
```
```suggestion

If you're looking to contribute on a fun project, we recommend trying to connect Dask dataframe up to an existing extension array like [cyberpandas](https://github.com/ContinuumIO/cyberpandas) or TODO.  

```

mrocklin replied:
Also, should we mention these sorts of projects at the beginning to add to the motivation?  Currently all of the new functionality we present is decimal.

mrocklin reviewed (commented):
This looks great.  I enjoyed the read.  I added a few tiny suggestions and a request to highlight some practical applications like cyberpandas earlier on.

TomAugspurger committed:
updates

TomAugspurger commented:
Thank, updated.

I reworked the "The Real Lesson" section a bit more to emphasize

1. Dask benefits directly from improvements to pandas (like the new nullable integer type)
2. 3rd party EAs (like cyberpandas) don't have to reimplement things from scratch, just to work with dask.

mrocklin commented:
@TomAugspurger OK to update the date in the filename and then merge this?

TomAugspurger committed:
Update date

TomAugspurger commented:
Sure. Done.

On Tue, Jan 22, 2019 at 9:51 AM Matthew Rocklin <notifications@github.com>
wrote:

> @TomAugspurger <https://github.com/TomAugspurger> OK to update the date
> in the filename and then merge this?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/dask/dask-blog/pull/10#issuecomment-456450316>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/ABQHItFeAPl32X8VZJfqaBvAVcuYEa7Rks5vFzOagaJpZM4aIajg>
> .
>
